client.go

```diff
@@ -177,7 +219,23 @@ func (r *Request) WithHeaders(headers http.Header) *Request {
 	return r
 }
 
+func (r *Request) WithOperationType(operation string) *Request {
+	switch strings.ToLower(operation) {
+	case "query":
```

Comment:
perhaps compress it to one case? I was surprised you can't do an assignment in a switch statement though
```golang
op := strings.ToLower(operation)
switch op {
case "query", "mutation", "subscription":
  r.OperationType = op
default:
  r.OperationType = "query"
}
```